src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -2937,6 +2938,63 @@ public async Task TestEngineSupportViewsWithoutKeyFieldsInConfigForMsSQL()
             }
         }
 
+        [TestMethod, TestCategory(TestCategory.COSMOSDBNOSQL)]
+        public async Task TestErrorMessageWithoutKeyFieldsInConfig()
+        {
+            Dictionary<string, object> dbOptions = new();
+            HyphenatedNamingPolicy namingPolicy = new();
+
+            dbOptions.Add(namingPolicy.ConvertName(nameof(CosmosDbNoSQLDataSourceOptions.Database)), "graphqldb");
+            dbOptions.Add(namingPolicy.ConvertName(nameof(CosmosDbNoSQLDataSourceOptions.Container)), "dummy");
+            dbOptions.Add(namingPolicy.ConvertName(nameof(CosmosDbNoSQLDataSourceOptions.Schema)), "custom-schema.gql");
+            DataSource dataSource = new(DatabaseType.CosmosDB_NoSQL,
+                GetConnectionStringFromEnvironmentConfig(environment: TestCategory.COSMOSDBNOSQL), Options: dbOptions);
+            Entity entity = new(
+                Source: new("EntityName", EntitySourceType.Table, null, null),
+                Rest: new(Enabled: false),
+                GraphQL: new("", ""),
+                Permissions: new[] { GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                Relationships: null,
+                Mappings: null
+            );
+
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(dataSource, new(), new(), entity, "EntityName");
+
+            const string CUSTOM_CONFIG = "custom-config.json";
+
+            File.WriteAllText(
+                CUSTOM_CONFIG,
+                configuration.ToJson());
+
+            string[] args = new[]
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+        };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string query = @"{
+                    EntityName {
+                        items{
+                            id
+                            title
+                        }
+                    }
+                }";
+
+                object payload = new { query };
+
+                HttpRequestMessage graphQLRequest = new(HttpMethod.Post, "/graphql")
+                {
+                    Content = JsonContent.Create(payload)
+                };
+
+                ApplicationException ex = await Assert.ThrowsExceptionAsync<ApplicationException>(async () => await client.SendAsync(graphQLRequest));
+                Assert.AreEqual(ex.Message, "The entity 'Planet' was not found in the dab-config json");
```

Comment:
why is this test sending a GraphQL request? if the entity doesn't exist in the config, shouldn't there be a startup error? 

Alternatively, if the intent is to capture some sort of error that occurs during runtime, then erroring on a non-existent entity should be handled by HotChocolate which detects that the requested entity is not part of the GraphQL schema -> HTTP400 bad request/GraphQL error because request doesn't adhere to schema.

Reply:
It is not schema error, it is config error, if there is an entity which is defined in schema but not defined in config then, `Hotchocolate` wouldn't even know about it since schema has this information. It will fail when DAB check for the entity in the config.

Reply:
can you add more details to the test description then? there's no context in the test describing what mismatch exists between schema.gql and dab-config.json. The request is querying "EntityName", but the error message says "Planet". That's not really actionable by the end user because it's not immediately clear what the relation is (or isn't) between "EntityName" and "Planet"